src/clients/SpokePoolClient.ts

```diff
@@ -751,6 +769,15 @@ export class SpokePoolClient extends BaseAbstractClient {
           fill.relayExecutionInfo.updatedMessageHash = getMessageHash(event.args.relayExecutionInfo.updatedMessage);
         }
 
+        // Sanity check that this event is not a duplicate.
+        const duplicateFill = this.fills[fill.originChainId]?.find(
+          (f) => f.transactionHash === fill.transactionHash && f.logIndex === fill.logIndex
+        );
```

Comment:
We repeat this iterative searching enough that it could probably be factored out to a utility function - that might condense the code a bit. Something like:

```
const sameEvent = (a: SortableEvent, b: SortableEvent): boolean =>
  a.transactionHash === b.transactionHash && a.logIndex === b.logIndex;

// ...

const duplicateFill = this.fills[fill.originChainId]?.find(sameEvent);
```

Reply:
[duplicate event util](https://github.com/across-protocol/sdk/pull/884/commits/29b94649c4674a91b523f13f01989d1c9ff388b6)